### PR TITLE
Fix compute item missing-argument assertion

### PR DESCRIPTION
**Pull request by wthrowe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- Code has documentation and unit tests
- Private member variables have a trailing underscore
- Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- Header order:
  1. hpp corresponding to cpp (only in cpp files) or `tests/Unit/TestingFramework.hpp` (only in tests)
  2. Blank line (only in cpp files and tests)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- File lists in CMake are alphabetical
- Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- Mark objects `const` whenever possible
- Almost always `auto`, except with expression templates, i.e. `DataVector`
- All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- Prefix commits addressing PR requests with `fixup`. These commits are flagged
  by Travis so we do not accidentally merge them.
- Include what you use, prefer forward declarations in hpp files.
- Explicitly make numbers floating point, e.g. `2.` or `2.0` over `2`
- Use `Tensor`'s non-member get if the indices are constant expressions, e.g.
  `get<1>(tensor)`

### CODE REVIEW

File: src/DataStructures/DataBox/DataBox.hpp
Comment (nilsdeppe) [inline src/DataStructures/DataBox/DataBox.hpp]: A comment explaining what the different parts of this are doing would probably be helpful

File: src/DataStructures/DataBox/DataBox.hpp
Comment (nilsdeppe) [inline src/DataStructures/DataBox/DataBox.hpp]: I'm not sure this is still the best name for this variable. maybe something like `compute_tag_argument_edges`?

File: src/DataStructures/DataBox/DataBox.hpp
Comment (nilsdeppe) [inline src/DataStructures/DataBox/DataBox.hpp]: Let's only enable this check in debug mode

File: src/DataStructures/DataBox/DataBox.hpp
Comment (nilsdeppe) [inline src/DataStructures/DataBox/DataBox.hpp]: Wouldn't this also be triggered if there are two of the same base tags in the DataBox? If so, then the message should mention that

File: src/DataStructures/DataBox/DataBox.hpp
Comment (nilsdeppe) [inline src/DataStructures/DataBox/DataBox.hpp]: "IsOk" is not the most helpful variable name. Maybe "FoundComputeArgumentInBox"?

Comment (nilsdeppe) [reply src/DataStructures/DataBox/DataBox.hpp]: Since these are tags, maybe `ComputeTag` and `ArgumentTag`?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by wthrowe** [comment]
I found a few more places that needed similar warnings, so I'm just adding them to this PR.

**Comment by wthrowe** [comment]
There are some related failures in db::item_type (passing a base tag with either no list or a list that does not contain anything valid), but I can't find a good way to deal with them.

File: src/DataStructures/DataBox/DataBox.hpp
Comment (kidder) [inline src/DataStructures/DataBox/DataBox.hpp]: its

**Review by kidder — CHANGES_REQUESTED** [review]
squash and ping @nilsdeppe

**Review by nilsdeppe — DISMISSED** [review]

**Comment by kidder** [comment]
@wthrowe this just needs squashing your fixups

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@nilsdeppe should just need your approval

**Review by mar-celine — APPROVED** [review]

**Comment by kidder** [comment]
@nilsdeppe you approved before Will squashed, so we will just merge

**Pull request merged**
